src/homework/server/services/ItemManager.js

```diff
@@ -0,0 +1,136 @@
+const pokemonClient = require("../clients/PokemonClient.js");
+const {Item, Sequelize} = require("../db/models")
+const fs = require("fs").promises;
+class ItemManager {
+  constructor() {
+    this.taskList = [];
+    this.newItems = [];
+    this.newItemsForTable=[];
+
+  }
+
+  async addItem(item) {
+    this.newItems = [];
+    this.newItemsForTable=[];
+    this.taskList = await Item.findAll({raw:true});
+    const pokemonObj = await pokemonClient.checkByPokemonName(item);
+    const { isPokemon, arrOfPokemonsID } = this.isPokemon(item);
+    if (isPokemon || pokemonObj) {
+      let ArrWithoutDuplicates;
+      let pokemons;
+      try {
+       //  ArrWithoutDuplicates = this.getItemsToAdd(arrOfPokemonsID);
+        if (pokemonObj) {
+          pokemons=[pokemonObj];
+          const pokemonObjForTable = { isPokemon: true, item: pokemonObj.name, isDisplay: false, status: false }
+          this.newItemsForTable.push(pokemonObjForTable);
+          const pokemonObject= { isPokemon: true, item: pokemonObj, isDisplay: false, status: false }
+          this.newItems.push(pokemonObject);
+        } else {
+          pokemons = await pokemonClient.fetchPokemon(arrOfPokemonsID);
+          pokemons.forEach((pokemon) => {
+            const pokemonObjForTable = { isPokemon: true, item: pokemon.name, isDisplay: false, status: false }
+            this.newItemsForTable.push(pokemonObjForTable);
+            const pokemonObj= { isPokemon: true, item: pokemon, isDisplay: false, status: false }
+            this.newItems.push(pokemonObj);
+          })
+        }
+        this.taskList = this.taskList.concat(
+          pokemons.map((pokemon) => {
+            const obj = { isPokemon: true, item: pokemon, isDisplay: false, status: false }
+            return obj
+          })
+        );
+      } catch (exception) {
+        let res = "";
+        ArrWithoutDuplicates.forEach(elem => {
+          res += elem + " ";
+
+        });
+      }
+    } else {
+      this.taskList.push({ isPokemon: false, item: item ,isDisplay: false, status: false});
```

Comment:
dont use this arrays. initialize the array in the beginning of the method and not in the class level.you cant hold state of the DB in your app, that's the DB's job.
Just parse the tasks added and create in the DB